### ISSUE TITLE
Can't build on 1.6

### ISSUE THREAD

**Issue by yongqli** [rust-lang-nursery/regex#124]
Compiling regex_macros v0.1.21 (https://github.com/rust-lang-nursery/regex#6f428bc5)
    /Users/yongqli/.cargo/git/checkouts/regex-5ba7da685e6a7b39/master/regex_macros/src/lib.rs:510:36: 510:54 error: unresolved name `ast::PatWildSingle` [E0425]
    /Users/yongqli/.cargo/git/checkouts/regex-5ba7da685e6a7b39/master/regex_macros/src/lib.rs:510                 node: ast::PatWild(ast::PatWildSingle),
                                                                                                                                     ^~~~~~~~~~~~~~~~~~
    /Users/yongqli/.cargo/git/checkouts/regex-5ba7da685e6a7b39/master/regex_macros/src/lib.rs:507:19: 511:16 note: in this expansion of vec! (defined in <std macros>)
    /Users/yongqli/.cargo/git/checkouts/regex-5ba7da685e6a7b39/master/regex_macros/src/lib.rs:510:36: 510:54 help: run `rustc --explain E0425` to see a detailed explanation

`rustc` version is `rustc 1.6.0-dev` on`2015-11-01`

**Comment by ivan** [comment]
https://github.com/rust-lang/rust/commit/3468b8d42c0c5a8510a2810747221354dbbb8eff

**Comment by ivan** [comment]
I just noticed that this change has *not* yet landed in the latest (as of right now) `rust-nightly-x86_64-unknown-linux-gnu.tar.gz` in https://static.rust-lang.org/dist/2015-11-01/ (`rustc 1.6.0-nightly (1a2eaffb6 2015-10-31)`)

**Comment by jethrogb** [comment]
This particular issue is easily fixed by
```diff
diff --git a/regex_macros/src/lib.rs b/regex_macros/src/lib.rs
index 5fe05bb..52de6ed 100644
--- a/regex_macros/src/lib.rs
+++ b/regex_macros/src/lib.rs
@@ -507,7 +507,7 @@ fn exec<'t>(
             pats: vec!(P(ast::Pat{
                 id: ast::DUMMY_NODE_ID,
                 span: self.sp,
-                node: ast::PatWild(ast::PatWildSingle),
+                node: ast::PatWild,
             })),
             guard: None,
             body: body,
```
Which then reveals another error
```
src/lib.rs:531:48: 531:60 error: no method named `parse_expr` found for type `syntax::parse::parser::Parser<'_>` in the current scope
src/lib.rs:531     let entry = cx.expander().fold_expr(parser.parse_expr());
                                                              ^~~~~~~~~~~~
src/lib.rs:532:23: 532:33 error: the type of this value must be known in this context
src/lib.rs:532     let regex = match entry.node {
```

**Comment by BurntSushi** [comment]
Fixed by #128 in f76921
